### PR TITLE
DAOS-9847 test: Refactor pool/destroy.py

### PR DESCRIPTION
**Pull request by shimizukko**
- Remove `test_destroy_invalid_group()` because `test_destroy_wrong_group()` essentially does the same thing.
- Remove `test_destroy_withdata()` because `test_destroy_connected()` tests the base case. Writing data and redoing it doesn't add any value.
- DAOS-3836 is fixed. The commented blocks checks the pool files in the file system, so leave it from `validate_pool_creation()`, but remove it from places that don’t make sense.

All tags in Test-tag are for the tests in this file.

Skip-unit-tests: true
Test-tag: destroy_single destroy_multi destroy_single_loop destroy_multi_loop destroy_invalid_uuid destroy_invalid_label destroy_wrong_group destroy_connected force_destroy_connected
Signed-off-by: Makito Kano <makito.kano@intel.com>

### CODE REVIEW

File: src/tests/ftest/pool/destroy.py [hunk 1]
```diff
@@ -167,7 +165,9 @@ def test_destroy_single(self):
         """Test destroying a pool created on a single server.
 
         :avocado: tags=all,pr,daily_regression
-        :avocado: tags=pool,destroy,destroy_single
+        :avocado: tags=vm
+        :avocado: tags=pool
+        :avocado: tags=destroy,destroy_single
```
Comment (daltonbohning) [inline src/tests/ftest/pool/destroy.py:170]: While this file is being refactored, could we refactor the tags too? Personally, I think "destroy" and "destroy_single" aren't clear when taken out of the context of this file. I would recommend
```suggestion
        :avocado: tags=vm
        :avocado: tags=pool,pool_destroy
        :avocado: tags=pool_destroy_single
```
And likewise for other tests in this file. Then, all tests in this file could be ran with `Test-tag: pool_destroy`.
What's your opinion?

**Review by daltonbohning — APPROVED** [review]
LGTM, except optional tag refactoring

**Review by dinghwah — APPROVED** [review]
LGTM

**Review by daltonbohning — APPROVED** [review]

File: src/tests/ftest/pool/destroy.py (hunk 1 above)
Comment (shimizukko) [reply src/tests/ftest/pool/destroy.py]: That makes sense. I updated them. Thanks.

**Pull request merged**